pennylane/_qubit_device.py

```diff
@@ -761,13 +761,13 @@ def expval(self, observable, shot_range=None, bin_size=None):
 
         # exact expectation value
         if self.shots is None:
-            try:
-                eigvals = self._asarray(observable.eigvals, dtype=self.R_DTYPE)
-            except NotImplementedError as e:
+            eigvals = observable.eigvals
+            if eigvals is None:
+                # eigenvalue representation is not defined
                 raise ValueError(
                     f"Cannot compute analytic expectations of {observable.name}."
-                ) from e
-
+                )
+            eigvals = self._asarray(eigvals, dtype=self.R_DTYPE)
```

Comment:
We now assume that "not defined" is `None` instead of `NotImplementedError`.

Note that in standard situations like this we don't need to query if a representation is defined through some "registration", since if it is defined, we would like to use the result.